DEV_GUIDE.md

```diff
@@ -26,6 +27,22 @@ This document gives a detailed breakdown of the various build processes and opti
   * [Rendering documentation](#rendering-documentation)
 <!-- TOC -->
 
+## Project structure
+
+The module structure is as follows:
+
+* `kroxylicious-bom` the Bill Of Materials used by developers to conveniently declare a consistent set of Kroxylicious dependencies.
+* `krpc-code-gen` a code generator used to build the API module.
+* `kroxylicious-api` the APIs needed to write a filter.
+* `kroxylicious-unit-test` test utilities for filter authors.
+* `kroxylicious` the proxy runtime platform for hosting filters, including the networking code etc, but lacking a `main()` method
+* `kroxylicious-test-tools` ???
```

Comment:
I wasn't sure how this differs from the unit-test module.

Reply:
More focused on integration tests or things that actually exercise message flows. @robobario is that how you would describe test tools?

Reply:
Yeah, test-tools contains some heavier integration testing tools like:

1. a mock Kafka server
2. a low-level Kafka client that can send one RPC at a time
3. KroxyliciousTester which is able to start up a Kroxylicious instance and make it easier for users to obtain clients that are configured to talk to the proxy (and has integration with the mock kafka-server, or the junit extension so the proxy can work with both.

The problem is, since that depends on `kroxylicious` we can't use it to write tests in the `kroxylicious` module, so `unit-test` was split off, and it holds some testing tools that `kroxylicious` can use, like the code for loading Kafka messages from YAML files.

Reply:
Maybe we could use:

`kroxylicious-test-support`
`kroxylicious-integration-test-support`

*-test is a common patter but `kroxylicious-integration-test` starts sounding like a suite of tests.